### DISCUSSION TITLE
What's your FC?

### DISCUSSION THREAD

**Discussion by howey** [joeycastillo/Sensor-Watch#285] [Polls]
For those who have calibrated with nanosec, what are you seeing for a static frequency correction?

I just got a red board and have been trying to calibrate it with finetune/nanosec. My watch is running way faster than I would have expected. It seems to be gaining about 1 second per hour, and even after a few minutes of finetune the drift is perceivable. I'm calibrating a few times per day. FC is up to -29000 now and I'm just wondering what is normal.

**Comment by 814d3** [comment]
My classic Sensor Wach is set to 1614. I calibrated it like 4 times (during one/two months) and it has been staying very accurate for several weeks.

**Comment by Niehztog** [comment]
My classic Sensor Watch is at 2601 and runs quite accurate with that.

**Comment by howey** [comment]
I think what is happening in my case is that the calibration variable `freq_correction` is `int16_t`, and I am actually overflowing this because my clock is running so fast.

But even if I made `freq_correction` a `int32_t` I don't think this would solve my issue. From the datasheet, FREQCORR has a maximum value of 127, which corresponds to 122ppm. In my case, my clock runs fast by about 1 second per hour, which would be an error of about 278 ppm.

So I think the nanosec strategy of using FREQCORR alone is not sufficient for watches with really bad oscillators greater than around 122ppm (which I seem to have).

**Comment by wryun** [comment]
I have two sensor watch boards, one at ~1700 and one at ~2500. Sounds like yours is defective :(

**Comment by KlingonJane** [comment]
I have two red sensor watch lite boards. First running well at 278.  The second one at 1414, but still tuning in.
